### PR TITLE
move erddap snippet generation to browser

### PR DESCRIPTION
**Pull request by n-a-t-e**
This PR moves ERDDAP snippet generation  from the repo called 'cioos-yaml-to-erddap' into this repo.

See https://github.com/cioos-siooc/cioos-yaml-to-erddap - that repo will be removed once this PR is merged.

### CODE REVIEW

File: src/utils/recordToERDDAP.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  };
+  const attributesStr = Object.entries(attributes)
+    .filter(([, v]) => v)
+    .map(([k, v]) => `  <att name="${xmlescape(k)}">${v}</att>`)
+    .join("\n");
+
+  return `<addAttributes>
+<iso19115File>/your/waf/${
+    record.filename || getRecordFilename(record)
+  }.xml</iso19115File>
```
Comment (sjbruce) [inline src/utils/recordToERDDAP.js:73]: This section should be inside `dataset` but outside of `addattributes`

**Review by sjbruce — CHANGES_REQUESTED** [review]
Looks good aside from the position of the iso19115File tag, once that adjustment is made this should be good to go.

**Comment by sjbruce** [comment]
It may also be desirable to add XML encoding for text fields to handle various characters like ampersands, degree symbols, quotes and the like.  I don't think ERDDAP will freak out if they're missing, but it may affect how the text/html is rendered.

File: src/utils/recordToERDDAP.js (hunk 1 above)
Comment (n-a-t-e) [reply src/utils/recordToERDDAP.js:73]: Nice catch

**Comment by n-a-t-e** [comment]
> It may also be desirable to add XML encoding for text fields to handle various characters like ampersands, degree symbols, quotes and the like. I don't think ERDDAP will freak out if they're missing, but it may affect how the text/html is rendered.

See `xmlescape` on this line - https://github.com/cioos-siooc/metadata-entry-form/pull/214/files#diff-f6b4d8d8b46b269da8757e17bb63bc9a76ad0e36906abee6bd88861a3f22bfd6R67

**Comment by sjbruce** [comment]
Ah, I missed that.  Strange though, I've got a couple of datasets that have characters in them that aren't being escaped, the main one being the Wave Energy Research Centre.

**Comment by n-a-t-e** [comment]
Degrees and quotes etc shouldn't need escaping, just ampersands and angle brackets

**Comment by n-a-t-e** [comment]
> Right, I see that now in the docs. However, the WERC dataset still has 2 ampersands in the text that aren't being escaped. Maybe it's just something with my local copy?

Ah its true even angle brackets aren't being escaped, actually it looks like I am escaping the key instead of the value, fix coming soon, good catch

**Pull request merged**